### PR TITLE
network: disable LinkLocalAddressing on sit devices

### LINKED ISSUE #12547: networkctl shows tunnel as configuring when it is configured
**systemd version the issue has been seen with**
> systemd 242 (242.19-1-arch)
+PAM +AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid

**Used distribution**
 > Arch Linux Arm for ARMv6 (Raspberry Pi Zero W)

**Expected behaviour you didn't see**
> After creating a sit tunnel with a valid remote and local set to any networkctl should show it as configured.

**Unexpected behaviour you saw**
 > After creating a sit tunnel with a valid remote and local set to any networkctl show it as configuring.

**Steps to reproduce the problem**
Create the following configuration files, change \<ipv4-address\> and \<ipv6-address\> to something valid.

/etc/systemd/network/tun0.netdev
```
[Match]

[NetDev]
Name=tun0
Kind=sit

[Tunnel]
Local=any
Remote=<ipv4-address>
TTL=255
```

/etc/systemd/network/tun0.network
```
[Match]
Name=tun0

[Network]
Address=<ipv6-address>
```

/etc/systemd/network/wlan0.network
```
[Match]
Name=wlan0

[Network]
DHCP=yes
Tunnel=tun0
```

If the NetDev's Local property is changed from any to a valid IPv4 address networkctl shows it as configured.

### PR DESCRIPTION
**Pull request by yuwata**
Closes #12547.

cc @ssahani.

### CODE REVIEW

File: src/network/networkd-link.c [hunk 1]
```diff
@@ -121,7 +121,7 @@ bool link_ipv4ll_enabled(Link *link) {
         if (!link->network)
                 return false;
 
-        if (STRPTR_IN_SET(link->kind, "vrf", "wireguard"))
+        if (STRPTR_IN_SET(link->kind, "vrf", "wireguard", "sit"))
```
Comment (ssahani) [inline src/network/networkd-link.c]: This should be for all the tunnel ipip/gre/gre6/vti/vti6 etc. Please define a macro ```dev_is_tunnel``` something like this and call that here.
```
sit-test: Gained carrier
sit-test: Acquiring IPv4 link-local address
Assertion '!ether_addr_is_null(&ll->mac)' failed at ../src/libsystemd-network/sd-ipv4ll.c:228, function ipv4ll_start_internal(). Ignoring.
sit-test: Could not acquire IPv4 link-local address: Invalid argument

```

**Comment by yuwata** [comment]
@ssahani Thanks. Yeah, I guess so. But I was lazy about checking that. I will check that and update this.

**Comment by yuwata** [comment]
@ssahani I noticed that some tunnel netdevs support IPv6LL but not IPv4LL. So, I do not introduce `link_is_tunnel()`. PTAL.

File: test/test-network/systemd-networkd-tests.py [hunk 2]
```diff
@@ -578,14 +581,12 @@ def test_geneve(self):
...
 
     def test_ipip_tunnel(self):
-        self.copy_unit_to_networkd_unit_path('12-dummy.netdev', '25-ipip-tunnel.netdev', 'ipip.network',
-                                             '25-ipip-tunnel-local-any.netdev', '25-ipip-tunnel-remote-any.netdev')
-        self.start_networkd()
-
-        self.assertTrue(self.link_exits('dummy98'))
-        self.assertTrue(self.link_exits('ipiptun99'))
-        self.assertTrue(self.link_exits('ipiptun98'))
-        self.assertTrue(self.link_exits('ipiptun97'))
```
Comment (ssahani) [inline test/test-network/systemd-networkd-tests.py:588]: Why remove the assert ?

Comment (yuwata) [reply test/test-network/systemd-networkd-tests.py:588]: If `wait_online()` passes then the checks are redundant.

Comment (ssahani) [reply test/test-network/systemd-networkd-tests.py:588]: hmm is not that looking for   ```dummy98:degraded'``` that does not mean link not created ?

Comment (yuwata) [reply test/test-network/systemd-networkd-tests.py:588]: No. operstate 'degraded' means that the link exists and has only link local addresses, in this case, an ipv6ll address.

Comment (ssahani) [reply test/test-network/systemd-networkd-tests.py:588]: right thanks.

**Comment by ssahani** [comment]
LGTM

**Comment by yuwata** [comment]
@ssahani Thank you for your review.

**Pull request merged**